### ISSUE TITLE
UI scale level is different when running from interpreter than from script

### ISSUE THREAD

**Issue by tonyxwz** [matplotlib/matplotlib#15674]
### Bug report

**Bug summary**

When run interactively from python console or ipython, the size of the ui (buttons) and figure dpi are normal. However, if the same code is placed in a script and called from console or IDE (vscode), the zoom level is much larger.

I suspect it is related to my screen resolution (2160x1440)

**Code for reproduction**

Just first create a script and paste the following code inside, then run the code line by line in ipython. 

```python
import numpy as np
import matplotlib.pyplot as plt

fig, axes = plt.subplots(1, 2)

n = np.arange(100)
axes[0].plot(n, np.sin(n))

axes[1].plot(n, np.tanh(n))

plt.show()
```

**Actual outcome**

If your screen has high dpi, you should see two figures with different zoom levels.

![image](https://user-images.githubusercontent.com/16152581/68625633-eb47ff80-04d9-11ea-8363-37de8956b2b0.png)

**Expected outcome**
The zoom level should be the same.

**Matplotlib version**

  * Operating system: windows 10x64
  * Matplotlib version: 3.1.1
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: Python 3.7.4 (default, Aug  9 2019, 18:34:13) [MSC v.1915 64 bit (AMD64)] :: Anaconda, Inc. on win32, default channel
  * Jupyter version (if applicable):
  * Other libraries: numpy 1.16.5

**Comment by QuLogic** [comment]
> Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg

Is this true in _both_ places?

**Comment by tonyxwz** [comment]
> Is this true in *both* places?
@QuLogic Yes, both figures are using Qt5Agg as backend.

**Comment by timhoffm** [comment]
I see a different behavior with 3.1.1: If scaling is <= 125%, I get the small window. If scaling is >= 150%, I get the large window.

The different spacing of icons is addressed via #14697, which will be in 3.2.0 and 3.1.2.

There's additionally the yet open #15656.

@tonyxwz please recheck if you have a dependency on there screen scaling factor in windows.

**Comment by tacaswell** [comment]
https://doc.qt.io/qt-5/highdpi.html

I suspect that the difference between the two is due to some combination of who created the QApp (Matplotilb, IPython, or someone els) and what the environment variables are .
